pkg/tektonlog/tektonlog.go

```diff
@@ -211,91 +211,72 @@ func (t *TektonLogger) GetRunningBuildLogs(ctx context.Context, pa *v1.PipelineA
-				return false
-			}
-			return t1.Before(t2)
-		})
 
-		for _, pt := range podTimes {
-			podName := pt.name
-			if completedPods[podName] {
+		for _, stage := range stages {
+			podName := stage.name
+			stageName := stage.task
+			if completedStages[stageName] {
 				continue
 			}
-			stageName := pt.task
-			log.Logger().Infof("logging pod: %s for task %s", info(podName), stageName)
-
-			pod, err := t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
-			if err != nil && apierrors.IsNotFound(err) {
-				if pa.Spec.Status == v1.ActivityStatusTypeRunning {
-					pa.Spec.Status = v1.ActivityStatusTypeAborted
+			if stage.podExists {
+				log.Logger().Infof("logging pod: %s for task %s", info(podName), stageName)
+
+				pod, err := t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
+				if err != nil && apierrors.IsNotFound(err) {
+					if pa.Spec.Status == v1.ActivityStatusTypeRunning {
+						pa.Spec.Status = v1.ActivityStatusTypeAborted
+					}
+					return nil
+				}
+				if err != nil {
+					return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
 				}
-				return nil
-			}
-			if err != nil {
-				return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
-			}
 
-			err = t.getContainerLogsFromPod(ctx, pod, pa, buildName, stageName, out)
-			if err != nil {
-				return errors.Wrapf(err, "failed to get logs for pod %s", podName)
-			}
+				err = t.getContainerLogsFromPod(ctx, pod, pa, buildName, stageName, out)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get logs for pod %s", podName)
+				}
 
-			// let's reload the pod to see if it's completed or not (it should, now that we've streamed its logs)
-			pod, err = t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
-			if err != nil && apierrors.IsNotFound(err) {
-				if pa.Spec.Status == v1.ActivityStatusTypeRunning {
-					pa.Spec.Status = v1.ActivityStatusTypeAborted
+				err = pods.WaitForPodNameToBeComplete(t.KubeClient, t.Namespace, podName, 1*time.Second)
+				if err == nil {
+					completedStages[podName] = true
 				}
-				return nil
-			}
-			if err != nil {
-				return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
-			}
-			if pods.IsPodCompleted(pod) {
-				completedPods[podName] = true
+
+			} else if stage.skipped {
+				completedStages[stageName] = true
+				log.Logger().Infof("pod is skipped/failed for task: %s", stageName)
 			}
 		}
 
-		if len(completedPods) > 0 {
+		if len(completedStages) > 0 {
 			foundLogs = true
 		}
 
-		// if all pods completed lets terminate
-		if len(completedPods) == len(podTimes) {
-			break
+		// if all pods completed lets move out from the loop
+		if len(completedStages) == len(stages) {
+			loggedAllRunsForActivity = true
+		} else {
+			time.Sleep(time.Second)
+			log.Logger().Debugf("let's a wait second for next pod/task to start")
```

Comment:
minor, but you should log before sleeping ;-)

Reply:
fixed